Assets/Scripts/Communication/Communication.cs

```diff
@@ -102,7 +105,7 @@ public static void Send(byte[] buffer)
 
 	public static byte[] Recv()
 	{
-		if (m_Messages.Count == 0) {
+		if (m_Messages.Count <= 0) {
```

Comment:
I don't think so.  It was just one of those things where the exception was pointing around this area and this change didn't fix it since it was a different issue.  The <= 0 instead of == 0 is just a safer check even if < 0 shouldn't be possible.